packages/javascript-kernel/src/kernel.ts

```diff
@@ -233,61 +177,108 @@ export class JavaScriptKernel extends BaseKernel implements IKernel {
   }
 
   /**
-   * Execute code in the kernel IFrame.
+   * Load the worker.
+   *
+   * ### Note
+   *
+   * Subclasses must implement this typographically almost _exactly_ for
+   * webpack to find it.
+   */
+  protected initWorker(): Worker {
+    return new Worker(new URL('./comlink.worker.js', import.meta.url), {
+      type: 'module',
+    });
+  }
+
+  /**
+   * Initialize the remote kernel.
    *
-   * @param code The code to execute.
+   * @param options The options for the remote kernel.
+   * @returns The initialized remote kernel.
    */
-  protected _eval(code: string): string {
-    return this._evalFunc(this._iframe.contentWindow, code);
+  protected initRemote(
+    options: JavaScriptKernel.IOptions
+  ): IRemoteJavaScriptWorkerKernel {
+    const remote: IRemoteJavaScriptWorkerKernel = wrap(this._worker);
+    remote.initialize({ baseUrl: PageConfig.getBaseUrl() });
+    return remote;
   }
 
   /**
-   * Create a new IFrame
+   * Process a message coming from the JavaScript web worker.
    *
-   * @param iframe The IFrame to initialize.
+   * @param msg The worker message to process.
    */
-  protected async _initIFrame(): Promise<void> {
-    if (!this._iframe.contentWindow) {
+  private _processWorkerMessage(msg: any): void {
+    if (!msg.type) {
       return;
     }
-    this._evalFunc(
-      this._iframe.contentWindow,
-      `
-        console._log = console.log;
-        console._error = console.error;
-
-        window._bubbleUp = function(msg) {
-          window.parent.postMessage(msg);
-        }
 
-        console.log = function() {
-          const args = Array.prototype.slice.call(arguments);
-          window._bubbleUp({
-            "event": "stream",
-            "name": "stdout",
-            "text": args.join(' ') + '\\n'
-          });
-        };
-        console.info = console.log;
+    const parentHeader = msg.parentHeader || this.parentHeader;
 
-        console.error = function() {
-          const args = Array.prototype.slice.call(arguments);
-          window._bubbleUp({
-            "event": "stream",
-            "name": "stderr",
-            "text": args.join(' ') + '\\n'
-          });
-        };
-        console.warn = console.error;
-
-        window.onerror = function(message, source, lineno, colno, error) {
-          console.error(message);
-        }
-      `
-    );
+    switch (msg.type) {
+      case 'stream': {
+        const bundle = msg.bundle ?? { name: 'stdout', text: '' };
```

Comment:
maybe all the empty responses could be hoisted to  `emptyStreamResponse = Object.freeze({name: 'stdout': text: ''})`, e.g constants up top?